src/Microsoft.AspNetCore.Routing/Matching/DfaMatcherBuilder.cs

```diff
@@ -375,8 +377,30 @@ public override Matcher Build()
             }
 
             var candidates = CreateCandidates(node.Matches);
+
+            // Perf: most of the time there aren't any endpoint selector policies, create
+            // this lazily.
+            List<IEndpointSelectorPolicy> endpointSelectorPolicies = null;
+            if (node.Matches?.Count > 0)
```

Comment:
Note there was a bug here that I fixed. This was passing in a 'null' list of matches to `AppliesToEndpoints`.